app/javascript/components/student/editor/CodeEditor.tsx

```diff
@@ -1,23 +1,35 @@
 import React, { useRef } from 'react'
+import MonacoEditor from 'react-monaco-editor'
+import * as monacoEditor from 'monaco-editor/esm/vs/editor/editor.api'
 
 export function CodeEditor({ onSubmit }: { onSubmit: (code: string) => void }) {
-  const editor = useRef<HTMLTextAreaElement>(null)
+  const editorRef = useRef<monacoEditor.editor.IStandaloneCodeEditor>()
 
-  function handleSubmit(e: React.FormEvent<HTMLFormElement>) {
+  function handleSubmit(e: React.MouseEvent<HTMLButtonElement>) {
     e.preventDefault()
 
-    if (editor.current === null) {
-      return
-    }
+    onSubmit(editorRef.current?.getValue() || '')
```

Comment:
Does this mean we're allowing submitting empty code?

Reply:
(I think we should)

Reply:
There's an empty string check before we submit so this is fine. I wrote it like this as the `onSubmit` function expects a string.

I agree with @SleeplessByte, though. We should allow empty code :)

Reply:
> We should allow empty code :)
> (I think we should)

Why?

Reply:
Because there is no difference between `" "` or `""` from a invalid code standpoint. Requiring a character to be able to see the first test case is meh. 

Alternative: show the test file.